### PR TITLE
fix error in types and cancel dir pkg under dir go

### PR DESCRIPTION
**Pull request by m3ngyang**
This is a sub pr of pr #565 
reorganize the structure of project.

### CODE REVIEW

File: go/apis/paddlepaddle/v1alpha1/types.go [hunk 1]
```diff
@@ -36,7 +36,10 @@ type TrainingJob struct {
 // TrainingJobSpec is the spec for a TrainingJob resource
 type TrainingJobSpec struct {
 	// General job attributes.
-	Image             string              `json:"image,omitempty"`
+	Image string `json:"image,omitempty"`
+	// If you want to use the hostnetwork instead of container network
+	// portmanager is necessary.
```
Comment (Yancey1989) [inline go/apis/paddlepaddle/v1alpha1/types.go]: Do we have a link to introduce the **portmanager**? The user may not know what's a **portmanager** is.

File: go/scripts/update-codegen.sh [hunk 2]
```diff
@@ -31,7 +31,7 @@ echo ${CODEGEN_PKG}
 #                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
 #                  instead of the $GOPATH directly. For normal projects this can be dropped.
 ${CODEGEN_PKG}/generate-groups.sh "deepcopy,client,informer,lister" \
- github.com/PaddlePaddle/cloud/go/pkg/client github.com/PaddlePaddle/cloud/go/pkg/apis \
+ github.com/PaddlePaddle/cloud/go/client github.com/PaddlePaddle/cloud/go/apis \
```
Comment (typhoonzero) [inline go/scripts/update-codegen.sh]: I see no `go/pkg/client` directory in current develop branch, is this really needed? Or can you add this when the package is committed.

Comment (m3ngyang) [reply go/scripts/update-codegen.sh]: @typhoonzero dir `client` will be generated for go test

File: go/apis/paddlepaddle/v1alpha1/types.go (hunk 1 above)
Comment (m3ngyang) [reply go/apis/paddlepaddle/v1alpha1/types.go]: @Yancey1989 ok, let me add a link

File: go/scripts/update-codegen.sh (hunk 2 above)
Comment (m3ngyang) [reply go/scripts/update-codegen.sh]: And I think it is strange only to change `pkg/apis`.

Comment (typhoonzero) [reply go/scripts/update-codegen.sh]: I see, is `generate-groups.sh "deepcopy,client,informer,lister"` generating the template code for later development?

Comment (m3ngyang) [reply go/scripts/update-codegen.sh]: Yes, if the structs of types change, the tools (client, deepcopy and so on) needs to be re-generated.

**Pull request merged**